iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/vector/HiveDeleteFilter.java

```diff
@@ -142,7 +143,7 @@ public HiveBatchContext next() {
 
           @Override
           public void close() throws IOException {
-            srcIterator.close();
+            IOUtils.close(srcIterator, deleteInputIterable, deleteOutputIterable);
```

Comment:
looks good to me, however I would suggest an additional minor refactor:
deleteInputIterable, deleteOutputIterable are defined outside of new CloseableIterable<HiveBatchContext>. in the filterBatch, can you move them as final fields into the CloseableIterable, emphasizing that it owns these iterables, so it's assumed to close them also
otherwise, their closure makes sense